x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportDeleteInferenceEndpointAction.java

```diff
@@ -0,0 +1,165 @@
+
+    private final ModelRegistry modelRegistry;
+    private final InferenceServiceRegistry serviceRegistry;
+    private static final Logger logger = LogManager.getLogger(TransportDeleteInferenceEndpointAction.class);
+
+    @Inject
+    public TransportDeleteInferenceEndpointAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        ModelRegistry modelRegistry,
+        InferenceServiceRegistry serviceRegistry
+    ) {
+        super(
+            DeleteInferenceEndpointAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            DeleteInferenceEndpointAction.Request::new,
+            indexNameExpressionResolver,
+            EsExecutors.DIRECT_EXECUTOR_SERVICE
+        );
+        this.modelRegistry = modelRegistry;
+        this.serviceRegistry = serviceRegistry;
+    }
+
+    @Override
+    protected void masterOperation(
+        Task task,
+        DeleteInferenceEndpointAction.Request request,
+        ClusterState state,
+        ActionListener<AcknowledgedResponse> masterListener
+    ) {
+        SubscribableListener.<ModelRegistry.UnparsedModel>newForked(modelConfigListener -> {
+            // Get the model from the registry
+
+            modelRegistry.getModel(request.getInferenceEndpointId(), modelConfigListener);
+        }).<Boolean>andThen((listener, unparsedModel) -> {
+            // Validate the request & issue the stop request to the service
+
+            if (request.getTaskType().isAnyOrSame(unparsedModel.taskType()) == false) {
+                // specific task type in request does not match the models
+                listener.onFailure(InferenceExceptions.mismatchedTaskTypeException(request.getTaskType(), unparsedModel.taskType()));
+                return;
+            }
+
+            if (request.isDryRun()) {
+                masterListener.onResponse(
+                    new DeleteInferenceEndpointAction.Response(
+                        false,
+                        InferenceProcessorInfoExtractor.pipelineIdsForResource(state, Set.of(request.getInferenceEndpointId()))
+                    )
+                );
+                return;
+            } else if (request.isForceDelete() == false
+                && endpointIsReferencedInPipelines(state, request.getInferenceEndpointId(), listener)) {
+                    return;
+                }
+
+            var service = serviceRegistry.getService(unparsedModel.service());
+            if (service.isPresent()) {
+                service.get().stop(request.getInferenceEndpointId(), listener);
+            } else {
+                listener.onFailure(
+                    new ElasticsearchStatusException("No service found for model " + request.getInferenceEndpointId(), RestStatus.NOT_FOUND)
```

Comment:
Same question as below, found this in a second pass :)

Reply:
Good catch!